### DISCUSSION TITLE
Add  classList on View base

### DISCUSSION THREAD

**Discussion by NathanaelA** [NativeScript/rfcs#20] [1. Ideas & Discussions]
One of the items I created way back in NS1 days was adding "classList" to the base view classes so that you can do:
`View.classList.toggle("hidden")` or `View.classList.add("blue");`   --  The current NS theme's also re-implements this same code and I've seen a couple other plugins re-implement this.    https://developer.mozilla.org/en-US/docs/Web/API/Element/classList -- I would be willing to merge my code from my plugin into core so that this is supported on all elements that have a "class" property.
The actual functions that `classList` support are documented here as classList is a DomTokenList object:
- https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList

This is one of the few utility functions that I find so helpful as you can easily add/remove/toggle classes on any element which can be very helpful in highlighting errors, flow, or even starting/stopping animations...

**Comment by farfromrefug** [comment]
What s the overhead of that on ViewBase? I tend to never use  css anymore because it is much faster. Thus my question, what s the overhead in code and perfs of adding classList?

**Reply by rigor789** [reply]
Likely minimal if even measurable. To my understanding, it would just be a method that manipulates the existing class string (not sure if that's what the plugins do).

**Reply by NathanaelA** [reply]
That is correct, it wouldn't add any overhead.   Basically it is a helper class so you don't have to do any string manipulation for CSS when changing it up.  
- If you aren't using CSS it won't effect you in the slightest. 
- If you are using CSS and aren't using the "new" helper functions it also won't effect you in the slightest. 
- If you are using the helper functions, it will be faster than using the current method of re-assigning a string  with new classes or removed classes (i.e. string manipulations) as the classList will instead effect the lower level Map directly making changing classes assigned to the view much faster.   :grinning:
